solarforecastarbiter/datamodel.py

```diff
@@ -0,0 +1,302 @@
+# coding: utf-8
+"""
+Data classes and acceptable variables as defined by the SolarForecastArbiter
+Data Model document. Python 3.7 is required.
+"""
+
+from dataclasses import dataclass, field
+import datetime
+
+
+import pandas as pd
+
+
+ALLOWED_VARIABLES = {
+    'surface_temperature': 'degC',
+    'surface_wind_speed': 'm/s',
+    'ghi': 'W/m^2',
+    'dni': 'W/m^2',
+    'dhi': 'W/m^2',
+    'poa_global': 'W/m^2',
+    'relative_humidity': '%',
+    'ac_power': 'MW',
+    'dc_power': 'MW',
+}
+
+
+@dataclass(frozen=True)
+class GetItem(object):
+    """Add a __getitem__ method for dict like access"""
+    def __getitem__(self, key):
+        return self.__dict__[key]
+
+
+@dataclass(frozen=True)
+class Site(GetItem):
+    """
+    Class for keeping track of Site metadata.
+
+    Parameters
+    ----------
+    name : str
+        Name of the Site, e.g. Desert Rock
+    latitude : float
+        Latitude of the Site in decimal degrees north of the equator,
+        e.g. 36.62373
+    longitude : float
+        Longitude of the Site in decimal degrees east of the
+        prime meridian, e.g. -116.01947
+    elevation : float
+        Elevation of the Site in meters above mean sea level, e.g. 1007
+    timezone : str
+        IANA timezone of the Site, e.g. Etc/GMT+8
+    network : str, optional
+        Measurement network name, e.g. SURFRAD
+    well_known_text: str, optional
+        Describes a geometric area for a Site which may be physically extended,
+        e.g. a polygon over a city for a Site that describes many distributed
+        generation PV systems.
+    extra_parameters : dict, optional
+        The extra parameters may be used by forecasters when
+        implementing other PV models. The framework does not provide
+        a standard set of extra parameters or require a particular
+        format – these are up to the site owner.
+    """
+    name: str
+    latitude: float
+    longitude: float
+    elevation: float
+    timezone: str
+    network: str = ''
+    well_known_text: str = ''
+    extra_parameters: dict = field(default_factory=dict)
```

Comment:
If we aren't enforcing the use of json, yaml or csv in extra parameters, perhaps extra_parameters should just be a string?